### ISSUE TITLE
Gas metrics tool

### ISSUE THREAD

**Issue by DanielRX** [juanfranblanco/vscode-solidity#113]
I was wondering if it would be possible to use the output from solc-js / solc to give gas estimates in the same way https://github.com/kisstkondoros/codemetrics does it.

If it's something you would be willing to support, I can put together a PR with some pointers to the gas estimate output in this ext.

**Comment by juanfranblanco** [comment]
So the idea is to provide a way to visualise and estimate the gas cost of a specific function or region within the code base on the ast / opcodes of that function?

That will be very useful as a way to review and optimise the code.

**Comment by rocky** [comment]
Others who have looked at solc's gas estimates say that too often it is a bit off. Especially if there are loops - however most solidity code doesn't have loops. 

That said, of course saving information from solc is a good idea. On the upcoming Mythril Platform integration we are doing, expect to see information from solc getting stored locally.

**Comment by DanielRX** [comment]
@rocky I completely agree, but my personal attempts at improving them have been delayed by other work. (A fixed length loop with a finite gas cost within the loop should be bounded by loopCount*gasCostInLoop + loopIndexUpdateCosts, AFAIK)

@juanfranblanco  So much like remix (where the estimated gas cost is shown on hover) we could add some form of tooltip or info the the functions to display the estimates from solc.

**Comment by notsegfault** [comment]
+1

**Comment by davidtperk** [comment]
+1

**Comment by clayrisser** [comment]
+1

**Comment by pinoli** [comment]
+1

**Comment by cyborgdennett** [comment]
+1 please make it happen

**Comment by Aziz87** [comment]
+1

**Comment by cameel** [comment]
Just a head's up - we're actually planning to remove the estimator from the compiler (https://github.com/ethereum/solidity/issues/13566) because it can't give satisfactory answers in many cases and it's unlikely to ever be able to. We also don't think it gets much use in practice.

**Comment by juanfranblanco** [comment]
Thanks @cameel ! So i guess is a no no, or it could be used an evm to simulate the gas.
